src/Entity/TerritorialCouncil/TerritorialCouncilMembership.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+namespace App\Entity\TerritorialCouncil;
+
+use Algolia\AlgoliaSearchBundle\Mapping\Annotation as Algolia;
+use App\Entity\Adherent;
+use App\TerritorialCouncil\Exception\TerritorialCouncilMembershipRoleException;
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Table(name="territorial_council_membership")
+ * @ORM\Entity
+ *
+ * @Algolia\Index(autoIndex=false)
+ */
+class TerritorialCouncilMembership
+{
+    public const TERRITORIAL_COUNCIL_MANAGER = 'MANAGER';
+    public const TERRITORIAL_COUNCIL_MEMBER = 'MEMBER';
+
+    public const ROLES = [
+        self::TERRITORIAL_COUNCIL_MANAGER,
+        self::TERRITORIAL_COUNCIL_MEMBER,
+    ];
+
+    /**
+     * @var int|null
+     *
+     * @ORM\Id
+     * @ORM\Column(type="integer", options={"unsigned": true})
+     * @ORM\GeneratedValue
+     */
+    private $id;
```

Comment:
can we just use EntityIdentityTrait here too ?

i am sure we will need to display pages, in frontend, of memberships (in a future manager controller or idk)